src/main/java/org/folio/linked/data/service/search/InstanceSearchServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package org.folio.linked.data.service.search;
+
+import static java.lang.String.format;
+import static java.util.Objects.isNull;
+import static java.util.stream.Collectors.joining;
+
+import feign.FeignException;
+import java.util.Collection;
+import java.util.Objects;
+import lombok.RequiredArgsConstructor;
+import org.folio.linked.data.client.SearchClient;
+import org.folio.linked.data.domain.dto.SearchResponseTotalOnly;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class InstanceSearchServiceImpl implements InstanceSearchService {
+
+  private static final String ID_NOT_EQUALS = "id <> \"%s\"";
+  private static final String LCCN_EQUALS = "lccn==\"%s\"";
+
+  private final SearchClient searchClient;
+
+  @Override
+  public ResponseEntity<SearchResponseTotalOnly> searchByLccn(Collection<String> lccn) {
+    return search(getLccnQuery(lccn));
+  }
+
+  @Override
+  public ResponseEntity<SearchResponseTotalOnly> searchByLccnExcludingId(Collection<String> lccn, String instanceId) {
+    var lccnQuery = getLccnQuery(lccn);
+    if (isNull(instanceId)) {
+      return searchByLccn(lccn);
+    }
+    return search(format("%s and %s", lccnQuery, format(ID_NOT_EQUALS, instanceId)));
+  }
+
+  private ResponseEntity<SearchResponseTotalOnly> search(String query) {
+    try {
+      return searchClient.searchInstances(query);
+    } catch (FeignException.NotFound e) {
+      return new ResponseEntity<>(new SearchResponseTotalOnly().totalRecords(0L), HttpStatus.NOT_FOUND);
+    }
+  }
+
+  private String getLccnQuery(Collection<String> lccnCol) {
+    var orLccnQuery = lccnCol.stream()
+      .filter(Objects::nonNull)
+      .map(lccn -> format(LCCN_EQUALS, lccn))
+      .collect(joining(" or "));
+    return format("(%s) and (%s)", orLccnQuery, excludeSuppressedRecords());
```

Comment:
the logic was copied from https://github.com/folio-org/ui-quick-marc/blob/master/src/hooks/useValidation/validators.js#L445